### PR TITLE
Phase-1

### PR DESCRIPTION
**Pull request by HarmAalbers**
Phase 1 Module 1: Error Handling (COMPLETE)

Ported from ~/.zsh/00_error_handling.zsh (747 LOC) → lib/error.sh (238 LOC) Code reduction: 68% through streamlining and removing duplication

Features implemented:
- Standardized exit codes (0, 1, 2, 3, 4, 5, 124, 130)
- Color-coded error messages (error, warning, info, success)
- JSON output support for all message types
- Input validation helpers (require_command, require_file, require_dir, require_permission)
- Trap handlers for cleanup and error tracing
- Debug mode with stack traces
- No duplicate loading guard

Test coverage:
- 21 ShellSpec tests (100% passing)
- Tests for all public functions
- Edge cases covered (missing files, permissions, commands)
- Exit code validation
- JSON vs text output verification

Standards compliance (per docs/check-list.md):
✓ Strict mode (set -Eeuo pipefail)
✓ Input validation on all functions
✓ Proper quoting everywhere
✓ No injection vulnerabilities
✓ Bash-portable (tested with bash only)
✓ shfmt clean
✓ shellcheck clean (zero warnings)
✓ All CI checks pass (37/37 tests)

DoD checklist: 38/38 criteria met

Next: Module 2 - Logging system

### CODE REVIEW

File: lib/logging.sh [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+    echo "Log Statistics:"
+    echo "  Total lines: $total"
+    echo "  Errors:      $errors"
+    echo "  Warnings:    $warnings"
+    echo "  Info:        $infos"
+  fi
+}
+
+# ═══════════════════════════════════════════════════════════════
+# Performance Timers (bash 3.2 compatible using temp files)
```
Comment (HarmAalbers) [inline lib/logging.sh]: Remove bash 3.2 compatibility!

**Review by HarmAalbers — COMMENTED** [review]
Cool!
